ARCHITECTURE.md

```diff
@@ -0,0 +1,94 @@
+# Standard Design and Architecture
+
+At the time of writing, almost a year of exploratory and freestyle project history has past.
+Fortunately, it is not necessary for further understanding, so I'll spare you that.
+This document, though, lays out the design, architecture and direction of Standard.
+
+If the topics discussed herein are dear to you, please take it as an invitation to get involved.
+
+This design document can only be altered through an RFC process.
+
+## Overview
+
+Standard is a _kitchen sink_ to bootstrap and sustain an _efficient_ Software Delivery Lifecycle (SDLC) with the power of Nix and Flakes.
+_Kitchen sink_, here, is a colloquial word for _Horizontal Integration Framework_ which integrates _vertical_ tooling, that is designed around the linux philosophy of doing one thing and to it well.
+The integration target, thereby, is the _efficient_ SDLC end-to-end process for which we offer suitable and well-integrated tools and best practices.
+An _efficient_ SDLC is characterized by two things.
+Firstly, an adequate _lead time_ which is the amount of time it takes to set up an initial version of the software delivery pipeline.
+It needs to be "adequate" rather than "just fast", because it takes place in the context of a team and thus encompasses learning and onboarding activities.
+Rather than optimizing for speed, these need optimization for success, above all.k
+Secondly, an efficient SDLC is characterized by short _cycle times_ which is the amount of time it takes for a commit to be shipped to a production environment.
+Along this journey, we encounter aspects of the development, or in the broader sense, the _contribution_ environment.
+We also encounter the packaging pipeline that produces our distributable artifacts.
+We encounter Continuous Building & Integration, Continuous Deployment as well as Continuous Observability and Discovery.
+The goal of Standard is to optimize the critical path along this process to achieve superior _cycle times_ through the powers of Nix and Flakes when compared to any other similar framework that doesn't recognize the intrinsic value of _reproducability_.
+
+## Scope
+
+The SDLC end-to-end process can be subdivided in roughly three process regions with different overall shapes and characteristics.
+That being said, it is important to note, that the shapeshifting nature across these process regions is by no means a valid justification to _break up_ the end-to-end perspective along these not-actual-boundaries.
```

Comment:
Agreed! It's important to resist the readers urge to break up and modularize the process. A puzzle is not the same thing as a process. But it can be expressed more imperatively (and shorter).